app/views/components/_chart.html.erb

```diff
@@ -5,8 +5,30 @@
   table_direction ||= "horizontal"
   from ||= false
   to ||= false
+  percent_metric ||= false
   rows ||= []
   keys ||= []
+
+  if percent_metric
+    maximum_y = 100
+  else
+    current_max = 0
+    rows.each do |r|
+      max = r[:values].max
+      current_max = max.to_i unless current_max >= max.to_i
+    end
```

Comment:
How about
```ruby
row_maximums = rows.map { |row| row[:values].max }
overall_maximum = row_maximums.max
```

Reply:
Updated, also added a default value so the comparisons don't try to compare against null if no data is passed. Would appreciate if you have time to give it another look, @cbaines

Reply:
That looks good. With the `-1` default value. That looks like it will lead the `maximum_y` to be set of `-1`, is that an ok value?

Reply:
The charts/tables deliberately don't render when there's no row data, so it's irrelevant what the value is in that case, but this stops the comparison from exploding against a null value.

Reply:
Ah, I see, how about checking for this case earlier then, perhaps:
```ruby
  elsif row.any?
    row_maximums = rows.map { |row| row[:values].max }
    overall_maximum = row_maximums.max
```

Reply:
updated

Reply:
Great :+1: